motr/examples/example2.c

```diff
@@ -0,0 +1,253 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For any questions about this software or licensing,
+ * please email opensource@seagate.com or cortx-questions@seagate.com.
+ *
+ */
+
+/*
+ * Example Motr application to create index, put key&val into index,
+ * get val from index, and then delete the index.
+ */
+
+/*
+ * Please change the dir according to you development environment.
+ *
+ * How to build:
+ * gcc -I/work/cortx-motr -I/work/cortx-motr/extra-libs/galois/include \
+ *     -DM0_EXTERN=extern -DM0_INTERNAL= -Wno-attributes               \
+ *     -L/work/cortx-motr/motr/.libs -lmotr                            \
+ *     example2.c -o example2
+ *
+ * Please change the configuration according to you development environment.
+ *
+ * How to run:
+ * LD_LIBRARY_PATH=/work/cortx-motr/motr/.libs/                              \
+ * ./example2 172.16.154.179@tcp:12345:34:1 172.16.154.179@tcp:12345:33:1000 \
+ *         "<0x7000000000000001:0>" "<0x7200000000000001:64>" 12345670
+ */
+
+#include "motr/client.h"
+#include "lib/trace.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <errno.h>
+
+enum {
+	KV_COUNT = 10,
+	KV_LEN   = 32,
+};
+
+static int index_launch_wait(struct m0_entity *e,
+			     struct m0_op    **ops,
+			     int               rc,
+			     int              *sm_rc)
+{
+	if (rc == 0) {
+		m0_op_launch(ops, 1);
+		rc = m0_op_wait(ops[0],
+				    M0_BITS(M0_OS_FAILED,
+					    M0_OS_STABLE),
+				    M0_TIME_NEVER);
+		printf("operation rc: %i\n", ops[0]->op_rc);
+		if (sm_rc != NULL) {
+			/* Save retcodes. */
+			*sm_rc = ops[0]->op_rc;
+		}
+	} else
+		printf("operation rc: %i\n", rc);
+	m0_op_fini(ops[0]);
+	m0_op_free(ops[0]);
+	ops[0] = NULL;
+	m0_entity_fini(e);
+	return M0_RC(rc);
+}
+
+int index_create(struct m0_container *container, struct m0_uint128 *fid)
+{
+	struct m0_op   *ops[1] = { NULL };
+	struct m0_idx   idx;
+	int             rc;
+
+	m0_idx_init(&idx, &container->co_realm, fid);
+
+	rc = m0_entity_create(NULL, &idx.in_entity, &ops[0]);
+	rc = index_launch_wait(&idx.in_entity, ops, rc, NULL);
+
+	printf("index create rc: %i\n", rc);
+	return rc;
+}
+
+int index_delete(struct m0_container *container, struct m0_uint128 *fid)
+{
+	struct m0_op   *ops[1] = { NULL };
+	struct m0_idx   idx;
+	int             rc = 0;
+
+	m0_idx_init(&idx, &container->co_realm, fid);
+
+	rc = m0_entity_open(&idx.in_entity, &ops[0]) ?:
+	     m0_entity_delete(&idx.in_entity, &ops[0]) ?:
+	     index_launch_wait(&idx.in_entity, ops, rc, NULL);
```

Comment:
op_fini()/free() won't be called here in case of an error.

Reply:
fixed.